airbyte-integrations/connectors/destination-s3/src/test/java/io/airbyte/integrations/destination/s3/S3DestinationAssumeRoleTest.java

```diff
@@ -0,0 +1,145 @@
+import com.amazonaws.services.identitymanagement.model.ListAttachedRolePoliciesRequest;
+import com.amazonaws.services.identitymanagement.model.ListRolePoliciesRequest;
+import com.amazonaws.services.identitymanagement.model.Policy;
+import io.airbyte.cdk.integrations.destination.s3.S3BaseChecks;
+import io.airbyte.cdk.integrations.destination.s3.S3DestinationConfig;
+import io.airbyte.cdk.integrations.destination.s3.S3DestinationConfigFactory;
+import io.airbyte.cdk.integrations.destination.s3.credential.S3AssumeRoleCredentialConfig;
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus.Status;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.Temporal;
+import java.time.temporal.TemporalUnit;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+public class S3DestinationAssumeRoleTest {
+  @Test
+  void testFailsWithNoEnvCredentials() {
+    S3Destination dest = new S3Destination(new S3DestinationConfigFactory(), Collections.emptyMap());
+    assertEquals(Status.FAILED, dest.check(S3DestinationTestUtils.getAssumeRoleConfig()).getStatus());
+  }
+
+  @Test
+  void testPassesWithAllCredentials() {
+    S3Destination dest = new S3Destination(new S3DestinationConfigFactory(), S3DestinationTestUtils.getAssumeRoleInternalCredentials());
+    assertEquals(Status.SUCCEEDED, dest.check(S3DestinationTestUtils.getAssumeRoleConfig()).getStatus());
+  }
+
+  @Test
+  void testFailsWithWrongExternalId() {
+    Map<String, String> envWithBadExternalId = new HashMap<>(S3DestinationTestUtils.getAssumeRoleInternalCredentials());
+    envWithBadExternalId.put("AWS_ASSUME_ROLE_EXTERNAL_ID", "dummyValue");
+    S3Destination dest = new S3Destination(new S3DestinationConfigFactory(), envWithBadExternalId);
+    assertEquals(Status.FAILED, dest.check(S3DestinationTestUtils.getAssumeRoleConfig()).getStatus());
+  }
+
+  private static final String REVOKE_OLD_TOKENS_POLICY = """
+         {
+           "Version": "2012-10-17",
+           "Statement": {
+             "Effect": "Deny",
+             "Action": "*",
+             "Resource": [
+                 "arn:aws:s3:::airbyte-integration-test-destination-s3/*",
+                 "arn:aws:s3:::airbyte-integration-test-destination-s3"
+             ],
+             "Condition": {
+               "DateLessThan": {"aws:TokenIssueTime": "%s"}
+             }
+           }
+         }
+         """;
+  private static final String POLICY_NAME_PREFIX = S3DestinationAssumeRoleTest.class.getSimpleName() + "Policy-";
+  private static final String ASSUMED_ROLE_NAME = "s3_acceptance_test_iam_assume_role_role";
+
+  @Test
+  @Timeout(value = 1, unit = TimeUnit.HOURS)
+  void testAutomaticRenewal() throws IOException, InterruptedException {
```

Comment:
can you add a comment in the code explaining what this test does? my understanding is
* get creds using the assume role config
* wipe out other policies from this test on that role (what happens if two people run tests at the same time?)
* run an upload with the creds
* create a policy to deny older token (which invalidates our current token)
* refresh the token + run an upload